mailpoet/lib/EmailEditor/Integrations/Core/Renderer/Blocks/Columns.php

```diff
@@ -44,16 +54,17 @@ private function getBlockWrapper(array $parsedBlock, SettingsController $setting
       <!--[if mso | IE]><table align="center" border="0" cellpadding="0" cellspacing="0" style="width:' . $width . ';" width="' . $width . '"><tr><td style="font-size:0px;mso-line-height-rule:exactly;"><![endif]-->
       <div style="margin-top:' . $marginTop . ';max-width:' . $width . ';padding-left:' . $layoutPaddingLeft . ';padding-right:' . $layoutPaddingRight . ';">
         <table
+          class="' . $classes . '"
           align="center"
           border="0"
           cellpadding="0"
           cellspacing="0"
           role="presentation"
-          style="background:' . $backgroundColor . ';background-color:' . $backgroundColor . ';max-width:' . $width . ';width:100%;"
+          style="' . esc_attr($settingsController->convertStylesToString($colorStyles)) . ';max-width:' . $width . ';width:100%;"
         >
           <tbody>
             <tr>
-              <td style="font-size:0px;background:' . $backgroundColor . ';background-color:' . $backgroundColor . ';padding-left:' . $paddingLeft . ';padding-right:' . $paddingRight . ';padding-bottom:' . $paddingBottom . ';padding-top:' . $paddingTop . ';text-align:left;">
+              <td style="font-size:0px;padding-left:' . $paddingLeft . ';padding-right:' . $paddingRight . ';padding-bottom:' . $paddingBottom . ';padding-top:' . $paddingTop . ';text-align:left;">
```

Comment:
You used the `esc_attr` function in the previous class. I a bit wonder why it's not also used here. Maybe we can create a ticket to unify all renderers in this way. What do you think?

Reply:
I addressed [this particular line in a follow-up PR](https://github.com/mailpoet/mailpoet/pull/5450/commits/caceefe0bac052c791c28dcc7210a8767adc06e8#diff-dcddd58f27adb701f13eac70e8f89fc681684e1d45d39df71e77f8d73ca7af23R66) but there are more unescaped places.
I'm glad you asked. I was thinking about adding a ticket oft this so I'll prepare a ticket. IMO we should add proper escaping to all the places where we build the HTML.